src/backend/utils/fmgr/fmgr.c

```diff
@@ -869,10 +871,14 @@ fmgr_security_definer(PG_FUNCTION_ARGS)
 		 * We could be calling either a regular or a set-returning function,
 		 * so we have to test to see what finalize flag to use.
 		 */
-		pgstat_end_function_usage(&fcusage,
-								  (fcinfo->resultinfo == NULL ||
-								   !IsA(fcinfo->resultinfo, ReturnSetInfo) ||
-								   ((ReturnSetInfo *) fcinfo->resultinfo)->isDone != ExprMultipleResult));
+		
+		if (!pltsql_pgstat_function_check_hook ||
+		    (fcache->prokind != PROKIND_PROCEDURE && !CALLED_AS_TRIGGER(fcinfo)) ||
+		    ((&fcusage)->fs != NULL && (*pltsql_pgstat_function_check_hook)(fcinfo)))
+			pgstat_end_function_usage(&fcusage,
```

Comment:
is it possible to put the hook inside pgstat_end_function_usage sp that we do not need to check it for every caller?

Reply:
pgstat_end_function_usage does not contain enough information for the hook function. We need proc OID, kind, etc which are not available there.